src/operator/convolution-inl.h

```diff
@@ -42,6 +43,7 @@ struct ConvolutionParam : public dmlc::Parameter<ConvolutionParam> {
   bool no_bias;
   dmlc::optional<int> cudnn_tune;
   bool cudnn_off;
+  bool force_nd_im2col;
```

Comment:
im2col algorithm has two versions: 1. 2-D version; 2. N-D version. By default, 2-D images will choose the first version, while force_nd_im2col=True forces the program to choose N-D version for 2-D images. This is used to verify the correctness of N-D version's implementation. Can we treat it as a hidden option from users?